lib/epochs/index.js

```diff
@@ -530,46 +432,150 @@ function getGroupInit(ssb, groupId, cb) {
+  const keys = epochs.map((epoch) => epoch.secret.toString('hex')).sort()
 
-    // case 4.4 - same membership
-    if (isSameSet(members1, members2)) {
-      preferredEpoch = tieBreak(tips)
-    }
+  const winningKey = Buffer.from(keys[0], 'hex')
 
-    // case 4.5 - one group has a membership that's a subset of other
-    else if (isSubsetOf(members1, members2)) {
-      preferredEpoch = tips[0]
-    } else if (
-      isSubsetOf(members2, members1)
-    ) {
-      preferredEpoch = tips[1]
-    }
-    // case 4.6 - groups have overlapping membership, but disjoint
-    // case 4.7 - disjoint membership (no overlap!)
+  return epochs.find((epoch) => epoch.secret.equals(winningKey))
+}
+
+function BuildPreferredEpoch(ssb, groupId) {
+  return function buildPreferredEpoch(reduce, cb) {
+    const tips = getTipsFromTangleReduce(reduce)
+
+    var preferredEpoch
+    if (tips.length === 1) preferredEpoch = tips[0]
+    else if (tips.length === 2) {
+      const [members0, members1] = tips.map((t) => t.members)
+
+      // case 4.4 - same membership
+      if (isSameSet(members0, members1)) {
+        preferredEpoch = tieBreak(tips)
+      }
+
+      // case 4.5 - one group has a membership that's a subset of other
+      else if (isSubsetOf(members0, members1)) {
+        preferredEpoch = tips[0]
+      } else if (isSubsetOf(members1, members0)) {
+        preferredEpoch = tips[1]
+      }
+
+      // case 4.6 - groups have overlapping membership, but disjoint
+      else if (
+        intersection(members0, members1).size > 0
+      ) {
+        // choose one for now,
+        preferredEpoch = tieBreak(tips)
+        // but also kick off resolution
+        fixDisjointEpochsLater(ssb, groupId) // <<< DELAYED SIDE EFFECTS!
+      }
+
+      // case 4.7 - disjoint membership (no overlap!)
 
+      // prettier-ignore
+      else return cb(Error('Membership case not handled yet'))
+    } else return cb(Error(`case of ${tips.length} tips not handled yet`))
+
+    if (preferredEpoch.members) delete preferredEpoch.members
+
+    cb(null, preferredEpoch)
+  }
+}
+
+// similar to getTipEpochs but a little slower because loading the membership of the epochs
+// NOTE could optimise to just load the membership of ONLY the tips later :shrug:
+function getTipEpochsWithMembers(ssb, groupId, cb) {
+  if (cb === undefined)
+    return p(getTipEpochsWithMembers).call(this, ssb, groupId)
+
+  const opts = { getters: Getters(ssb) }
+  epochsReduce(ssb, groupId, opts, (err, reduce) => {
     // prettier-ignore
-    else return cb(Error('Membership case not handled yet'))
-  } else return cb(Error(`case of ${tips.length} tips not handled yet`))
+    if (err) return cb(clarify(err, 'Failed to resolve epoch @tangle/reduce'))
+
+    const tips = getTipsFromTangleReduce(reduce)
+    return cb(null, tips)
+  })
+}
+
+function fixDisjointEpochsLater(ssb, groupId) {
+  const timeout =
+    ssb?.config?.tribes2?.disjointResloveDelay ??
```

Comment:
spelling mistake. and this should be mentioned in the readme

Reply:
and can't this reuse the existing config options we have? https://github.com/ssbc/ssb-tribes2/#config

Reply:
we could also reuse this function https://github.com/ssbc/ssb-tribes2/blob/165967eb2333ff89ed8a6e36e95be1c4b194e6f7/listeners.js#L34-L40

Reply:
I'm gonna say "no not mentioning in README" as it's a testing config.
I'm also gonna keep it simple and not reuse that function as this is serving atm

Reply:
Fixed spelling! THanks